examples/plot_credit_loan_decisions.py

```diff
@@ -0,0 +1,1293 @@
+    print(
+        f"For epsilon {epsilon}, ExponentiatedGradient learned {len(models)} inner models"
+    )
+
+
+# %%
+# Here, we can see all the inner models learned for each value of
+# :code:`epsilon`. With the :code:`ExponentiatedGradient` model, we specify an
+# :code:`epsilon` parameter that represents the maximal disparity in our fairness
+# metric that our final model should satisfy. For example, an :code:`epsilon=0.02`
+# means that the training value of the *equalized odds difference* of the
+# returned model is at most :code:`0.02` (if the algorithm converges).
+
+# %%
+# Reviewing inner models of ExponentiatedGradient
+# ===============================================
+# In many situations due to regulation or other technical restrictions, the
+# randomized nature of :code:`ExponentiatedGradient` algorithm may be undesirable. In
+# addition, the multiple inner models of the algorithm introduce challenges for
+# model interpretability. One potential workaround to avoid these issues is to
+# select one of the inner models and deploy it instead.
+#
+# In the previous section, we trained multiple :code:`ExponentiatedGradient` models
+# at different :code:`epsilon` levels and collected all the inner models learned by
+# this process. However, some of these inner models are better performance in
+# both their *balanced error rate* and *equalized odds difference*. We will
+# filter out these *"dominated"* models and plot the performance trade-offs of
+# the remaining models.
+
+# %%
+def is_pareto_efficient(points):
+    """
+    Filters a NumPy Matrix to remove rows that are strictly dominated
+    by another row in the matrix. Stricly dominated means the all the row values are greater than
+    the values of another row.
+
+    Parameters:
+        Points: NumPy array (NxM) of model metrics. Assumption that smaller values for metrics are preferred.
+
+    Returns:
+        Boolean Array: Nx1 boolean mask representing the non-dominated indices.
+    """
+    n, m = points.shape
+    is_efficient = np.ones(n, dtype=bool)
+    for i, c in enumerate(points):
+        if is_efficient[i]:
+            is_efficient[is_efficient] = np.any(
+                points[is_efficient] < c, axis=1
+            )
+            is_efficient[i] = True
+    return is_efficient
+
+
+# %%
+def filter_dominated_rows(points):
+    """
+    Removes rows from a DataFrame that are monotonically dominated by another
+    row in the DataFrame.
+
+    Parameters:
+        Points: DataFrame where each row represents the summarized performance (balanced accuracy, fairness metric)
+        of an inner model.
+
+    Returns:
+        Pareto_mask: Boolean mask representing indices of input DataFrame that are not monotonically dominated.
+        Masked_Dataframe: DataFrame with dominated rows filtered out.
+    """
+    pareto_mask = is_pareto_efficient(points.to_numpy())
+    return pareto_mask, points.loc[pareto_mask, :]
+
+
+# %%
+def aggregate_predictor_performances(
```

Comment:
We should replace the chunk of the code starting roughly here by a call to:
https://fairlearn.org/v0.8/api_reference/fairlearn.metrics.html#fairlearn.metrics.plot_model_comparison

Reply:
We're currently using `aggregate_predictor_performances` and `model_performance_sweep` function to determine which inner models of the `ExponentiatedGradient` are Pareto-dominated. If we were to replace this chuck of code with a call to `plot_model_comparison`, would we show the performance of all the inner models or still just the non-dominated ones.

I think we could replace the lines from *1031*-*1054* with this new function, if we really wanted.

Reply:
You are right that things are intertwined... I was hoping that we could somehow just use is_pareto_efficient and filter_dominated_rows, but not the rest. I see two options: (1) leave things as they are, (2) use the plot_model_comparison (on all of the models, without the restricting to pareto optimal ones)--was it a mess to look at all of them?

Reply:
I believe there are 35 inner models total (18 for one value of `epsilon`, 17 for the other), so it would be a bit cluttered with all the inner models.

Reply:
ok. let's leave as is for now then.